src/web-auth/captcha.js

```diff
@@ -374,7 +381,8 @@ function render(auth0Client, element, options, callback) {
   return {
     reload: load,
     getValue: getValue,
-    runArkose: runArkose
+    requiresTrigger: requiresTrigger,
```

Comment:
If we would change this to:

```
requiresTrigger: function() {
  return provider === ARKOSE_PROVIDER;
}
```

And then inside the `load` function, we do:

```
provider = challange.provider;
```

Where `provider` is defined right above the load function (inside the render):

```
let provider: string;
```

Wouldn't that allow us to drop the `window.` usage that's being introduced here?

Reply:
However, I wonder if it would make sense if we drop both requiresTrigger and triggerCaptcha, but do the following in the `done()` inside render:

```js
done(challenge.provider === ARKOSE_PROVIDER ? triggerCaptcha : undefined	)
```

That would change the consuming API, but ensures that we pass along a trigger to the consumer and they can access it once the captcha is loaded.

```js
var captcha = webAuth.renderCaptcha(
        document.querySelector('.captcha-container'),
        null,
        (trigger) => // Here trigger is undefined unless arkose
);
```

Reply:
The main issue here is that `renderCaptcha` has a sync and async aspect:

- it returns an object, which is a sync API
- it has a callback, which is for async API's

Mixing these two is something we should avoid. I do understand this was already part of the SDK before, but I am not sure we should continue adding to it, instead we should focus on adding to the async API (which is why I mention to pass a trigger to the callback).

Reply:
Any thoughts on this @pmalouin?

Reply:
going with callback based async approach after offline discussion